src/DataModel.php

```diff
@@ -143,4 +144,26 @@ public function filterRow(array $condition)
 		return $this->data_source->filterOne($condition)->getData();
 	}
 
+
+	/**
+	 * @param array $columns
+	 * @return array|bool
+	 */
+	public function aggregationRow(array $columns)
+	{
+		$hasAggregation = FALSE;
+		/** @var Column $column */
+		foreach ($columns as $column) {
+			if ($column->hasAggregationFunction()) {
+				$hasAggregation = TRUE;
+				$aggregationFunction = $column->getAggregationFunction();
+				$this->data_source->addAggregationColumn($aggregationFunction->getAggregationType(), $aggregationFunction->getColumn());
+			}
+		}
+		if (!$hasAggregation) {
+			return FALSE;
```

Comment:
Ah, you mean we should return empty array from every datasource, and if datasource doesn't return array then we should throw exception. That can work.